### PR TITLE
Fix the deprecation warning in the DMD DUB package + build with `-de` + CI check

### PR DESCRIPTION
**Pull request by wilzbach**
`de` is used already for the DMD build, tools, phobos etc. It was simply forgotten for the new DMD DUB package. The idea of building with `-de` is that whenever a new deprecation message gets introduced it should be ensured that all core packages can use the replacement. It's also bad for the image if user see deprecation warnings in a library they use.

### CODE REVIEW

**Review by jacob-carlborg — APPROVED** [review]

**Review by JinShil — APPROVED** [review]

File: dub.sdl [hunk 1]
```diff
@@ -60,6 +60,7 @@ subPackage {
   versions "GC"
   versions "NoMain"
   sourcePaths "src/dmd"
+  buildRequirements "disallowDeprecations"
```
Comment (MartinNowak) [inline dub.sdl]: Please don't hardcode this into the package, it incurs immediate breakage on deprecations for people that cannot update their compiler.

**Comment by MartinNowak** [comment]
> The idea of building with -de is that whenever a new deprecation message gets introduced it should be ensured that all core packages can use the replacement.

Yes, we should test that `-de` compiles and works for us, but we shouldn't enforce package users to use it as well, as this removes the deprecation phase out of deprecations.

File: ci.sh [hunk 2]
```diff
@@ -97,9 +97,13 @@ test_dub_package() {
     if [ "${DMD:-dmd}" == "gdmd" ] ; then
         echo "Skipping DUB examples on GDC."
     else
+        local abs_build_path="$PWD/$build_path"
         pushd test/dub_package
         for file in *.d ; do
+            # build with host compiler
             dub --single "$file"
+            # build with built compiler (~master)
+            dub --single --compiler="${abs_build_path}" "$file"
```
Comment (MartinNowak) [inline ci.sh]: You could either add sub-configurations to dub.sdl or build with DFLAGS to check that -de succeeds.

Comment (wilzbach) [reply ci.sh]: Ok.

**Comment by RazvanN7** [comment]
LGTM

**Pull request merged**